mutiny-core/src/sql/glue.rs

```diff
@@ -0,0 +1,609 @@
+            .collect::<Vec<_>>();
+
+        assert_eq!(api_test_rows[2].id, 3);
+        assert_eq!(api_test_rows[2].name, "test3");
+        assert!(api_test_rows[2].nullable.is_none());
+        assert!(api_test_rows[2].is);
+    } else {
+        panic!("Expected Payload::Select");
+    }
+
+    // do some bytea tests
+    glue.execute("CREATE TABLE IF NOT EXISTS mutiny_kv (key TEXT PRIMARY KEY, val BYTEA NOT NULL)")
+        .await
+        .unwrap();
+
+    let key_value_pairs = vec![
+        (vec![1, 2, 3], vec![4, 5, 6]),
+        (vec![7, 8, 9], vec![10, 11, 12]),
+    ];
+    let serialized_data = bincode::serialize(&key_value_pairs).unwrap();
+    let hex_serialized_data = hex::encode(serialized_data);
+
+    glue.execute("INSERT INTO mutiny_kv (key, val) VALUES ('storage', X'')")
+        .await
+        .unwrap();
+    let select_query = "SELECT val FROM mutiny_kv WHERE key = 'storage'";
+    let mut result = glue.execute(select_query).await.unwrap();
+
+    if let Payload::Select { rows, .. } = result.pop().expect("should get something") {
+        if let Some(row) = rows.first() {
+            if let Value::Bytea(hex_string) = &row[0] {
+                let serialized_data = hex::decode(hex_string).unwrap();
+                assert!(serialized_data.is_empty());
+            } else {
+                panic!("Expected bytea");
+            }
+        }
+    } else {
+        panic!("Expected Payload::Select");
+    }
+
+    let update_query = format!(
+        "UPDATE mutiny_kv SET val = X'{}' WHERE key = 'fedimint_storage'",
+        hex_serialized_data
+    );
+    glue.execute(&update_query).await.unwrap();
+
+    let select_query = "SELECT val FROM mutiny_kv WHERE key = 'fedimint_storage'";
+    let mut result = glue.execute(select_query).await.unwrap();
+
+    if let Payload::Select { rows, .. } = result.pop().expect("should get something") {
+        if let Some(row) = rows.first() {
+            if let Value::Bytea(binary_data) = &row[0] {
+                let retrieved_pairs: Vec<(Vec<u8>, Vec<u8>)> =
+                    bincode::deserialize(binary_data).unwrap();
+
+                assert_eq!(retrieved_pairs, key_value_pairs);
+            } else {
+                panic!("Expected bytea");
+            }
+        }
+    } else {
+        panic!("Expected Payload::Select");
+    }
+}
+
+#[cfg(test)]
+#[cfg(not(target_arch = "wasm32"))]
+mod tests {
+    use super::*;
+    use gluesql::prelude::MemoryStorage;
+    use tokio;
+
+    #[tokio::test]
+    async fn basic_glue_tests() {
+        let storage = MemoryStorage::default();
+        let mut glue = Glue::new(storage);
+        run_basic_glue_tests(&mut glue).await;
+    }
+
+    #[tokio::test]
+    async fn create_glue_storage_tests() {
+        // TODO reactivate
+        // create_glue_and_fedimint_storage().await;
+    }
+
+    #[tokio::test]
+    async fn create_glue_storage_value_tests() {
+        // TODO reactivate
+        // create_glue_storage_value().await;
```

Comment:
can we enable these now?

Reply:
No, glue can't run on a server. I may need to just delete these.